server/functions/index.js

```diff
@@ -70,53 +77,114 @@ exports.sendNotificationNearbyOnNewFavor = functions.firestore
                     var latUser = user.location.latitude;
                     var longUser = user.location.longitude;
                     var distance = distanceInKm(longFav, latFav, longUser, latUser);
-                    if (distance < maxDistance) {
+                    if (distance < maxDistance && user.id !== posterId) {
                         usersIds.push(user.notificationId)
                     }
                 });
 
-                sendMessage(message, usersIds);
+                sendMulticastMessage(message, usersIds);
 
             })
             .catch((err) => {
                 console.log('Error getting documents', err);
             });
     });
 
-// send notification to the user who accepted the favor for any update
+// send new favor notification to users on updates
 exports.sendNotificationOnUpdate = functions.firestore
     .document('/favors/{favorId}')
-    .onWrite((change) => {
+    .onUpdate((change) => {
 
-        // get new favor that has just been posted
-        const newFavor = change.data();
-        var accepterId = newFavor.accepter;
-        var favorTitle = newFavor.title;
-        var usersIds = [];
+        const oldFavor = change.before.data();
+        const updatedFavor = change.after.data();
+        const oldStatus = oldFavor.statusId;
+        const newStatus = updatedFavor.statusId;
 
-        const message = {
-            notification: {
-                title: 'Favor + ' + favorTitle + ' has been updated',
-                body: 'Check out the new details',
-            },
-            tokens: usersIds
-        };
+        const favorTitle = updatedFavor.title;
 
-        // go through all the users
-        db.collection('/users').get()
-            .then((snapshot) => {
-                snapshot.forEach((doc) => {
-                    var user = doc.data();
-                    var userID = user.id;
-                    if (userID === accepterId) {
-                        usersIds.push(user.notificationId);
-                    }
-                });
+        const userIds = updatedFavor.userIds;
 
-                sendMessage(message, usersIds);
+        if (userIds == null) {
+            return;
+        }
 
-            })
-            .catch((err) => {
-                console.log('Error getting documents', err);
-            });
+        // that's always present
+        const requesterId = userIds[0];
+
+        // get accepterId, if it exists
+        var accepterId = "";
+        if (userIds.length > 1) {
+            accepterId = userIds[1];
+        }
+
+        var titleToSend = "";
+        var userReceiver = "";
+
+        // send notification if new status
+        if (oldStatus !== newStatus) {
+
+            //console.log('Status has changed');
+
+            // favor has been accepted, send notification to requester
+            if (newStatus === 1) {
```

Comment:
For clarity do you think we should use representations like final static int ACCEPTED = 1 ?

Reply:
Yes, that would be better. I just extracted global constants in Javascript.